bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/api/dto/clip2/helper/Setters.java

```diff
@@ -341,7 +349,60 @@ public static Resource setResource(Resource target, Resource source) {
                 targetTimedEffects.setDuration(duration);
             }
         }
-
         return target;
     }
+
+    /**
+     * Merge on/dimming/color fields from light and grouped light resources.
+     * Subsequent resources will be merged into the first one.
+     * Full state resources are not supported by this method.
+     */
+    public static Collection<Resource> mergeLightResources(Collection<Resource> resources) {
+        Map<String, Resource> resourceIndex = new HashMap<>();
+        Iterator<Resource> iterator = resources.iterator();
+        while (iterator.hasNext()) {
+            Resource resource = iterator.next();
+            String id = resource.getId();
+
+            if (resource.hasFullState()) {
+                throw new IllegalStateException("Resource " + id + " has full state, this is not expected");
+            }
+
+            Resource indexedResource = resourceIndex.get(id);
+            if (indexedResource == null) {
+                indexedResource = resource;
+                resourceIndex.put(id, indexedResource);
+            }
+
+            if (!LIGHT_TYPES.contains(resource.getType()) || !resource.hasHSBField()
+                    || resource.equals(indexedResource)) {
```

Comment:
Also I think you can remove the resource.equals() check (which is quite a dangerous check if someone would override the equals method in future), and instead insert a continue after line 374 ??

Reply:
Oh, I see you did that already..

Reply:
I agree it's dangerous and I didn't like it one bit. Initially I did `resource == indexResource`, but that triggered a SAT warning. When I tried to suppress that, I instead got a compiler warning about the suppresion. Since we didn't have an override, I made the dubious choice to use `equals`. I'm glad we can now remove it after the other simplifications, but I still would like a solution for this generic problem with SAT for future cases.